runtime/src/blueoil_image.cpp

```diff
@@ -0,0 +1,194 @@
+	throw "wrong channles != 1,3"; // XXX
+    }
+    const int clamped_x = clamp(x, 0, width-1);
+    const int clamped_y = clamp(y, 0, height-1);
+    const int scanlineSize = width * channels;
+    float *imagePtr = &(tensor.data[0]);
+    float *scanlinePtr = imagePtr + clamped_y * scanlineSize;
+    float *pixelPtr = scanlinePtr + clamped_x * channels;
+    return pixelPtr;
+}
+
+Tensor Tensor_CHW_to_HWC(Tensor &tensor) {
+    const int channels  = tensor.shape[0];
+    const int height = tensor.shape[1];
+    const int width  = tensor.shape[2];
+    if ((channels != 1) && (channels != 3)) {
+	throw "wrong channles != 1,3"; // XXX
+    }
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    int srcPlaneSize = width * height;
+    float *srcImagePtr = &(tensor.data[0]);
+    float *dstImagePtr = &(dstTensor.data[0]);
+    for (int y = 0 ; y < height ; y++) {
+	for (int x = 0 ; x < width ; x++) {
+	    float *srcPixelPtr0 = srcImagePtr + x + (y * height);
+	    for (int c = 0 ; c < channels ; c++) {
+		float *srcPixelPtr = srcPixelPtr0 + (c * srcPlaneSize);
+		*dstImagePtr = *srcPixelPtr;
+		dstImagePtr++;
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+Tensor Tensor_HWC_to_CHW(Tensor &tensor) {
+    int height = tensor.shape[0];
+    int width  = tensor.shape[1];
+    int channels  = tensor.shape[2];
+    if ((channels != 1) && (channels != 3)) {
+	throw "wrong channles != 1,3"; // XXX
+    }
+    Tensor dstTensor = Tensor::zeros({channels, height, width});
+    float *srcImagePtr = &(tensor.data[0]);
+    float *dstImagePtr = &(dstTensor.data[0]);
+    for (int c = 0 ; c < channels ; c++) {
+	float *srcPixelPtr = srcImagePtr + c;
+	for (int y = 0 ; y < height ; y++) {
+	    for (int x = 0 ; x < width ; x++) {
+		*dstImagePtr = *srcPixelPtr;
+		srcPixelPtr += 3;
+		dstImagePtr++;
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+/*
+ * Resize Image
+ */
+Tensor ResizeHorizontal(Tensor &tensor, const int width,
+			const enum ResizeFilter filter) {
+    if (filter != RESIZE_FILTER_NEAREST_NEIGHBOR) {
+	throw "unknown ResizeFilter:"; // XXX
+    }
+    const int srcHeight = tensor.shape[0];
+    const int srcWidth  = tensor.shape[1];
+    const int channels  = tensor.shape[2];
+    const int height = srcHeight;
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    float xScale = (float) width / (float) srcWidth;
+    int xSrcWindow = floorf(1/xScale);
+    xSrcWindow = 1; // Nearest
+    //xSrcWindow = (xSrcWindow < 2)? 2 :xSrcWindow; // Bi-Liner
+    for (int dstY = 0 ; dstY < height ; dstY++) {
+	for (int dstX = 0 ; dstX < width ; dstX++) {
+	    int srcX = (int) floorf(dstX/xScale);
+	    int srcY = dstY;
+	    for (int c = 0 ; c < channels ; c++) {
+		float v = 0.0;
+		float totalW = 0.0;
+		for (int x = -xSrcWindow ; x < xSrcWindow; x++){
+		    float *srcRGB = blueoil::image::Tensor_at(tensor,
+							      srcX + x, srcY);
+		    float d = fabs((float)x / (float)xSrcWindow);
```

Comment:
`std::fabs()` ?

And let's use `C++` casts not `C` ones.

`static_cast<float>` for this case.